### DISCUSSION TITLE
How to enable vitest strict type checking on function arguments whose type is a union of string literals

### DISCUSSION THREAD

**Discussion by dulimarta** [vuejs/pinia#2719] [Help and Questions]
The Pinia store below defines the `turn` function which takes a union of string literals as its parameter.

```
import { defineStore } from 'pinia';

export const useMyStore = defineStore('xyz', () => {
  function turn(d: 'left' | 'right'): boolean {
    return d === 'left';
  }
  return {
    turn,
  };
});
```

When 'turn' is invoked outside a test environment with a wrong argument, the TS compiler correctly indicates the issue:

```
const aStore = useMyStore()
aStore.turn('left')
aStore.turn('right')
aStore.turn('abc') // Flagged as an error

```

But when invoked with a wrong argument in a test environment, the TS compiler **does not** flag it as an error.

```
import { expect, test } from 'vitest';
import { createTestingPinia } from '@pinia/testing';
import { useMyStore } from '../src/mystore.js';
import { setActivePinia } from 'pinia';

test('union argument', () => {
  const testPinia = createTestingPinia({ stubActions: false });
  setActivePinia(testPinia);
  const aStore = useMyStore(testPinia);
  // turn('abc') is not flagged as an error
  expect(aStore.turn('abc')).toBeFalsy();
});
```

How do I enable stricter type checking inside vitest test environment?

**Comment by kevinramharak** [comment]
Can you provide some more details on your setup. How do you run the type checker when testing?
When I put your example code into the typescript playground it does flag it as an error:
https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgEwKYDNgDtUGUbSpwC+c6UEIcA5GNsAIbUDcAUK6gB6SxwDGELAGd4AVyGoAsgE98hOAF4UGbHgJRUACmqdpAL2oAaOJoCUigHyJWcMqKx8YwQXBiioWTcgBcNADYYMNRwAD40UMAA5gAWQaa+AEYQEAEMWNa2thpuHiiKCkrUAehBbLbENnDZ7ukIlbY5WIaVxGzEpmysAsLwDHIainDiUrLqWqasfWMAdI3axXGT-aizNdoRMYtThKse2gwJfNTmAPQncABifgyRkajIcAxCj+moUBRQ7KA88EhcYKhHMYYKgRCQyBQqNQAG7AEEiFisb7QX78DQMEEAFVBTiwkQACvQGODyJQaAABOhYRgneG4yKI5G8JASGAAQUcwGhqEJ1OJpFJUKpjERrDp2nsznSDCgkVEIFQWCCxjMlgy-EEYLpvMYgz46KxOOwBKJmhZbgSHKcmt86AYfgkJA6lVZVq5PNN2qJzts3TB2wGSmGMmWmi9fJ9cDOrjW1AOR3MwGeWAg8HQ11u90ezzScDeH0q-0BME0AZWczjh2OplmEAAQqgLvahNIzG1nUA

**Reply by dulimarta** [reply]
Thank you for your response. Your TS playground helped me find the issue with my `tsconfig.json`. For some reasons, I found that all the test files in our project are in the exclude list 😃 

```
{
  "exclude", ["node_modules", "**/__tests__/**"],
}
```

Removing the second entry above fixed the issue. 👍🏽
